src/Symfony/Component/Form/Extension/Core/Type/CountryType.php

```diff
@@ -35,15 +35,35 @@ public function configureOptions(OptionsResolver $resolver)
 
                 $choiceTranslationLocale = $options['choice_translation_locale'];
                 $alpha3 = $options['alpha3'];
+                $withEmoji = $options['with_emoji'];
+                $onlyEmoji = $options['only_emoji'];
 
-                return ChoiceList::loader($this, new IntlCallbackChoiceLoader(function () use ($choiceTranslationLocale, $alpha3) {
-                    return array_flip($alpha3 ? Countries::getAlpha3Names($choiceTranslationLocale) : Countries::getNames($choiceTranslationLocale));
+                if ($onlyEmoji && $withEmoji) {
+                    throw new LogicException(sprintf('Options "%s" and "%s" cannot be true together', 'with_emoji', 'only_emoji'));
+                }
+
+                return ChoiceList::loader($this, new IntlCallbackChoiceLoader(function () use ($choiceTranslationLocale, $alpha3, $withEmoji, $onlyEmoji) {
+                    $names = $alpha3 ? Countries::getAlpha3Names($choiceTranslationLocale) : Countries::getNames($choiceTranslationLocale);
+                    $choices = $names;
+                    if (true === $onlyEmoji) {
+                        foreach ($names as $name => $displayed) {
```

Comment:
the variable names are wrong, as keys are country code, not names. The name is what you called `$displayed`